internal/processors/token_transfer.go

```diff
@@ -0,0 +1,350 @@
+	return builder.WithCategory(category).
+		WithAccount(accountID).
+		WithAmount(amount).
+		WithAsset(asset, contractAddress).
+		WithLiquidityPool(poolID).
+		Build()
+}
+
+// createClaimableBalanceChange creates a state change for claimable balance interactions.
+// Includes the claimable balance ID to track which balance was created, claimed, or clawed back.
+func (p *TokenTransferProcessor) createClaimableBalanceChange(category types.StateChangeCategory, accountID, claimableBalanceID, amount string, asset *asset.Asset, contractAddress string, builder *StateChangeBuilder) types.StateChange {
+	return builder.WithCategory(category).
+		WithAccount(accountID).
+		WithAmount(amount).
+		WithAsset(asset, contractAddress).
+		WithClaimableBalance(claimableBalanceID).
+		Build()
+}
+
+// handleTransfer processes transfer events based on the operation type.
+// Different operations require different state change patterns:
+// - Claimable balances: single debit/credit with claimable balance ID
+// - Liquidity pools: single debit/credit with liquidity pool ID
+// - Regular transfers: debit/credit pair between accounts
+func (p *TokenTransferProcessor) handleTransfer(transfer *ttp.Transfer, contractAddress string, builder *StateChangeBuilder, operationType *xdr.OperationType) ([]types.StateChange, error) {
+	amount := transfer.GetAmount()
+	asset := transfer.GetAsset()
+
+	switch *operationType {
+	case xdr.OperationTypeCreateClaimableBalance:
+		// When creating a claimable balance, record debit from creator with CB ID
+		change := p.createClaimableBalanceChange(types.StateChangeCategoryDebit, transfer.GetFrom(), transfer.GetTo(), amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+
+	case xdr.OperationTypeClaimClaimableBalance:
+		// When claiming a claimable balance, record credit to claimer with CB ID
+		change := p.createClaimableBalanceChange(types.StateChangeCategoryCredit, transfer.GetTo(), transfer.GetFrom(), amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+
+	case xdr.OperationTypeLiquidityPoolDeposit:
+		// When depositing to LP, record debit from depositor with LP ID
+		change := p.createLiquidityPoolChange(types.StateChangeCategoryDebit, transfer.GetFrom(), transfer.GetTo(), amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+
+	case xdr.OperationTypeLiquidityPoolWithdraw:
+		// When withdrawing from LP, record credit to withdrawer with LP ID
+		change := p.createLiquidityPoolChange(types.StateChangeCategoryCredit, transfer.GetTo(), transfer.GetFrom(), amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+
+	case xdr.OperationTypeSetTrustLineFlags, xdr.OperationTypeAllowTrust:
+		// Skip events generated by these operations since they involve only an LP and Claimable Balance ID
+		return nil, nil
+
```

Comment:
Operations that pertain to liquidity pools, path payments and offers can all generate additional effects: modify funds on offers and liquidity pools. That's not entirely covered in this PR though and I don't think it should.

I think we could disable those operation types for now (as you did for OperationTypeAllowTrust and OperationTypeSetTrustLineFlags), and open a dedicated PR to handle them after we wrap this one. That way you'll simplify this PR and move this complexity to the next one, where you can focus solely on these specific cases.